assets/src/hooks/useSearchResultsByProperty.ts

```diff
@@ -0,0 +1,82 @@
+import { Socket } from "phoenix"
+import { PropertyLimits } from "../models/searchQuery"
+import { Loading, Ok } from "../util/fetchResult"
+import {
+  LimitedSearchResults,
+  useLimitedSearchResults,
+} from "./useSearchResults"
+import { useLocationSearchResults } from "./useLocationSearchResults"
+import { LocationSearchResult } from "../models/locationSearchResult"
+import { Vehicle, Ghost } from "../realtime"
+
+export type VehicleResultType =
+  | Ok<LimitedSearchResults<Vehicle | Ghost>>
+  | Loading
+  | null
+
+type LocationResultsType =
+  | Ok<LimitedSearchResults<LocationSearchResult>>
+  | Loading
+  | null
+
+export type SearchResultsByProperty = {
+  vehicle: VehicleResultType
+  operator: VehicleResultType
+  run: VehicleResultType
+  location: LocationResultsType
+}
```

Comment:
question: Would making this generic make it easier to share this code between autocomplete and search results without making the code too complex?

maybe something along the lines of this?
```suggestion
export type SearchPropertyResults<TVehicleResult, TLocationResult> = {
  vehicle: TVehicleResult
  operator: TVehicleResult
  run: TVehicleResult
  location: TLocationResult
}

export type SearchResultsByProperty = SearchPropertyResults<VehicleResultType, LocationResultsType>
export type AutocompleteResultsByProperty = SearchPropertyResults<VehicleResultType, LocationSuggestionsResultsType>
```

Reply:
Ooh I like that idea! I wonder if it might help to make a generic `SearchPropertyResult` type too so they can share loading / ok / null states? I'm not sure that autocomplete will need to keep the shape of `LimitedSearchResults` though... 

I think I'll hold off on adding generics here for now since I'm not quite seeing the full path to code sharing with autocomplete, but I support adding them as part of that work! 
```
type SearchPropertyResult<T> = Ok<LimitedSearchResults<T>> | Loading | null

export type SearchResultsByProperty<TVehicleResult, TLocationResult> = {
  vehicle: SearchPropertyResult<TVehicleResult>
  operator: SearchPropertyResult<TVehicleResult>
  run: SearchPropertyResult<TVehicleResult>
  location: SearchPropertyResult<TLocationResult>
}
```